### PR TITLE
fix(helm): use the same scc as olm

### PR DESCRIPTION
**Pull request by razvan**
# Description

See: https://github.com/stackabletech/issues/issues/624

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: deploy/helm/secret-operator/templates/roles.yaml [hunk 1]
```diff
@@ -67,11 +104,13 @@ rules:
       - podlisteners
     verbs:
       - get
+{{ if .Capabilities.APIVersions.Has "security.openshift.io/v1" }}
   - apiGroups:
     - security.openshift.io
     resourceNames:
-    - privileged
+      - stackable-secret-operator-scc
```
Comment (NickLarsenNZ) [inline deploy/helm/secret-operator/templates/roles.yaml:111]: Seems to have mixed indentation styles compared to the lines around it, but maybe that's just github mobile.

Comment (NickLarsenNZ) [reply deploy/helm/secret-operator/templates/roles.yaml:111]: Also, I think the PR description should explain how this is different to the other operators (using the rootv2 thing)

Comment (razvan) [reply deploy/helm/secret-operator/templates/roles.yaml:111]: I fixed the indentation thanks.

There is a [high-level note ](https://docs.stackable.tech/home/nightly/kubernetes/openshift#_security_context_constraints) in the documentation describing the different SCCs. Do you think it's enough for now?

Comment (NickLarsenNZ) [reply deploy/helm/secret-operator/templates/roles.yaml:111]: FYI, other sections have different style indentations for lists

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**